src/v/crypto/include/crypto/ossl_context_service.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "ssx/thread_worker.h"
+
+#include <memory>
+
+namespace crypto {
+using is_fips_mode = ss::bool_class<struct is_fips_mode_tag>;
+/**
+ * Service used to configure shard local OpenSSL library contexts.
+ * These contexts will be fetched by the OpenSSL library when performing
+ * cryptographic operations.
+ */
+class ossl_context_service final {
+public:
+    /**
+     * @brief Construct a new ossl context service object
+     *
+     * @param thread_worker Thread worker used to perform I/O operations outside
+     * of the Seastar context
+     * @param config_file Path to the OpenSSL config file
+     * @param module_path Path to the directory that contains the FIPS module
```

Comment:
Which one?  The config_file?  Sorry GH is showing me 4 lines :)

Reply:
The module_path is only valid if fips_mode::yes right?

Reply:
If we wanted to use the legacy provider we would need it... but I don't think we would (no need for MD2 or RC4)

In fact the config file really is only needed in FIPS mode (I have tests that show that starting up in non FIPS mode without a config works fine).

I'll come up with some thing

Reply:
maybe there is just `std::optional<fips_config>`?

Reply:
I kind of like keeping the flag.  Then I can check that both config file and module path are provided when in FIPS mode and error out if they're not provided.

Reply:
Yeah this is sort of an awkward thing to express in the typesystem in C++ and `std::variant<no_fips_tag, fips_config>` is awkward. I'll stop trying to be pedantic about the typesystem enforcing you can't do silly things (so you don't even need to handle errors of the required config not being present because the typesystem doesn't allow you to set the flag true without giving values). <rant over>

Reply:
Gotcha, I appreciate where you're coming from =). I guess part of me is trying to future proof for an event where we do need module_path or conf_File even when _not_ in FIPS mode.  This may come up when trying to add support for MD5 (which the FIPS module doens't provide but the base one does)